### ISSUE TITLE
Super Mario 3D Land doesn't open load file screen after completing World 8

### ISSUE THREAD

**Issue by digoserra** [Lime3DS/Lime-3DS-Emulator#198]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

2112 and older

### Description of Issue

I just completed Super Mario 3D Land's world 8 today. But after beating the game, pressing A on the title screen gives me a white screen. Music keeps going as normal but it doesn't open the load file screen anymore.

I did a bit of research and someone on Reddit said it happens because after finishing the game the save file gets some flashing stars to indicate it was a perfect game (with no white Tanooki Mario help), The freezing happens because of those flashing stars.

https://github.com/Lime3DS/Lime3DS/assets/15115624/15569267-2c4e-407c-9515-9c5939173b6e

### Expected Behavior

The load file screen loads as used to.

### Reproduction Steps

After finishing the game without using any special aid (P-Wings), just press A at the title screen.
[GameData.zip](https://github.com/Lime3DS/Lime3DS/files/15366839/GameData.zip)

### Log File

Not sure how to make one.

### System Configuration

CPU: Ryzen 5 4600G
GPU/Driver: Radeon RX 6600, driver version 24.3.1
RAM: 16 GB
OS: Windows 10 Home

**Comment by Riot20058** [comment]
Yes did test citra bug happened MMJ and last official ver going while before fixed it on lime3ds

**Comment by digoserra** [comment]
[lime3ds_log.zip](https://github.com/Lime3DS/Lime3DS/files/15371750/lime3ds_log.zip)

Found the log file.

**Comment by Riot20058** [comment]
> [lime3ds_log.zip](https://github.com/Lime3DS/Lime3DS/files/15371750/lime3ds_log.zip)
> 
> Found the log file.

Sorry sat it probably be while emulator bug thier in citra right see mainly focused UI and Pretendo support right now hope fixed game up see 3ds harder emulated might think runbow pocket very unstable

**Comment by ghost** [comment]
Lol, i remember being so mad the first time this bug happened to me 

For now, a workaround would be to get the invisibility leaf intentionally. I don't know how this bug was left in by the original team

**Comment by Jhon-ito** [comment]
This happened to me recently and I found a workaround that let's you get rid of the sparkling stars (which are causing the issue), continue your game right where you left off with no changes to the levels or your stats, and finally (if you want) restore your earned sparkling stars.
You will need a hex editor (HxD) and a save editor for the game (SGE), here it is: https://www.mediafire.com/file/c31kxtva5hstx5w/Super_Mario_Land_3D_SGE_v2b.exe/file

First you have to locate your save data for the game (GameData.bin), just open citra/lime and right click the mario 3d land icon and open the save directory, make a backup in case something goes wrong.
Open the .bin file with HxD, press ctrl+G and locate the following value according to your game file:
2B8F for file A
3512 for file B
3E95 for file C
Change that value (which should be 1C if you completed all levels up to world 8) to 5C, save and exit HxD. Now open your edited .bin file with the SGE tool then save and exit. you don't need to do anything else with the SGE tool, this step is only for the data to not get corrupted.
Now take your edited GameData.bin and replace the file on the citra/lime save directory.
Open the game, the sparkling stars will be gone and you can continue your playthrough. 

Optional: After you're done with the game you can restore your sparkling stars if you want, you just have to do the process again, take the GameData.bin file, open it with HxD, locate the value according to your game file, and change that value (which should be 7F if you completed all levels of the game) to 3F. Then open and save the game with the SGE tool and finally replace your edited .bin file on the citra/lime save directory. This will bring back your sparkling satrs but of course, if you try to open the game on citra/lime, you will have the same white screen problem. I suppose the only way to see them right now is if you open the game with that save file on an actual 3DS.

I did this all myself and it worked, and had no issues completing the rest of the game. Credits to this post that made me realise this could be done: https://gbatemp.net/threads/how-to-get-5-sparkly-stars-savefile-back-on-super-mario-3d-land.542645/

**Comment by ghost** [comment]
Really wish i could debug graphics and actually fix this but sadly i can't

**Comment by edoiya** [comment]
@Jhon-ito Thank you so much! It worked for me perfectly, you are the best.

**Comment by PabloMK7** [comment]
Already found a fix, but more work is needed. The fix will be added in release 2121. For now please use the fixes described here.

**Comment by jun-eau** [comment]
Can anyone confirm if this is fixed? Still running into this problem on version 2123.1.

**Comment by skoshy** [comment]
> Can anyone confirm if this is fixed? Still running into this problem on version 2123.1.

Still running into this on 2123 as well

**Comment by ALAiNKH** [comment]
> Mi è successo di recente e ho trovato una soluzione alternativa che ti permette di eliminare le stelle scintillanti (che causano il problema), continuare la partita esattamente da dove eri rimasto senza modifiche ai livelli o alle statistiche e infine (se vuoi) ripristinare le stelle scintillanti guadagnate. Avrai bisogno di un editor esadecimale (HxD) e di un editor di salvataggio per il gioco (SGE), eccolo: https://www.mediafire.com/file/c31kxtva5hstx5w/Super_Mario_Land_3D_SGE_v2b.exe/file
> 
> Per prima cosa devi trovare i dati di salvataggio del gioco (GameData.bin), apri Citra/Lime e clicca con il tasto destro sull'icona di Mario 3D Land, quindi apri la directory di salvataggio. Fai un backup nel caso in cui qualcosa vada storto. Apri il file .bin con HxD, premi ctrl+G e trova il seguente valore in base al tuo file di gioco: 2B8F per il file A 3512 per il file B 3E95 per il file C Modifica quel valore (che dovrebbe essere 1C se hai completato tutti i livelli fino al mondo 8) in 5C, salva ed esci da HxD. Ora apri il file .bin modificato con lo strumento SGE, salva ed esci. Non devi fare altro con lo strumento SGE, questo passaggio serve solo per evitare che i dati vengano corrotti. Ora prendi il tuo file GameData.bin modificato e sostituiscilo nella directory di salvataggio Citra/Lime. Apri il gioco, le stelle scintillanti saranno scomparse e potrai continuare la tua partita.
> 
> Facoltativo: una volta terminato il gioco, puoi ripristinare le tue stelle scintillanti se lo desideri. Devi solo ripetere la procedura, prendere il file GameData.bin, aprirlo con HxD, individuare il valore in base al file di gioco e modificarlo (che dovrebbe essere 7F se hai completato tutti i livelli del gioco) in 3F. Quindi apri e salva il gioco con lo strumento SGE e infine sostituisci il file .bin modificato nella directory di salvataggio citra/lime. Questo ripristinerà le tue stelle scintillanti, ma ovviamente, se provi ad aprire il gioco su citra/lime, avrai lo stesso problema della schermata bianca. Immagino che l'unico modo per vederle al momento sia aprire il gioco con quel file di salvataggio su un vero 3DS.
> 
> Ho fatto tutto da solo e ha funzionato, e non ho avuto problemi a completare il resto del gioco. Un ringraziamento a questo post che mi ha fatto capire che si poteva fare: https://gbatemp.net/threads/how-to-get-5-sparkly-stars-savefile-back-on-super-mario-3d-land.542645/

Devo provarlo. Sono la testimonianza che questo bug persiste con azahar. In pratica all'ai chiesi questo: su azahar mettendo il salvataggio bin di 3d land la schermata diventa bianca. Ci sono segnalazioni come questa, come la mia esperienza appena descritta? Risposta di copilot: Sì, ci sono segnalazioni identiche alla tua: Super Mario 3D Land su Azahar può mostrare una schermata bianca dopo il caricamento di un salvataggio .bin, soprattutto se il file proviene da una partita completata o modificata.

**Comment by PlayerGameSK** [comment]
> Can anyone confirm if this is fixed? Still running into this problem on version 2123.1.

Just got 2123.4-beta2 and still running into the same issue.

**Comment by Mr-Mamma-Mia** [comment]
I'm having this same problem myself, and I'm using 2124.3, I tried using the HxD thing, and went to the appropriate line but am not seeing a 1C value, mainly because I haven't cleatred all the levels as I skipped part of world 1 and world 4 via the pipes found in their second levels, so what would the value be if you have not cleared all the levels?

**Comment by Jhon-ito** [comment]
> I'm having this same problem myself, and I'm using 2124.3, I tried using the HxD thing, and went to the appropriate line but am not seeing a 1C value, mainly because I haven't cleatred all the levels as I skipped part of world 1 and world 4 via the pipes found in their second levels, so what would the value be if you have not cleared all the levels?

I literally tested random combinations untill I finally came across that solution so I don't know. Did you try to cahnge it to 5C and see if it messes up with your level progression? I'm almost sure that value in the one causing the sparkling satrs so maybe you should try testing some combinations too if 5C doesn't work properly. (remember always making a backup of your savefile).

**Comment by Mr-Mamma-Mia** [comment]
Well because it said to change the number associated with the save file, and I have save file 1, so it'd be 2B8F, and even then, I looked for it and the closest I found was 2B80, and there was a cell that had 1C, so I modified that to 5C, and when I switched the save that gets the white screen with the new save, it reads it as a brand new fresh save

**Comment by Jhon-ito** [comment]
I think 2B8F is still the one you have to modify for the sparkling stars problem, doesn't matter your value is different form 1C, try changing it to 5C and see what happens, also, remember to save the game with the SGE tool, or else the data will get corrupted and the game will just show a new fresh savefile.

**Comment by Mr-Mamma-Mia** [comment]
Well because I can't find the 2B8F row, and I also did resave it with SGE and it still showed a fresh save file when swapping the saves around

**Comment by Jhon-ito** [comment]
Well, sorry but I don't know what to tell you, pressing ctrl+G and typing 2B8F should guide you to the right value if you're using file A, maybe not having all levels unlocked afects the save file hex values more than I expected.

**Comment by Mr-Mamma-Mia** [comment]
Guess this is what I get for tryna 100 percent "and" speedrun

**Comment by Jhon-ito** [comment]
May I have a look of your savefile? If you don't mind sharing it, I got curious to see how it looks and why the workaround does not work.

**Comment by Mr-Mamma-Mia** [comment]
Well, when I tried uploading it on another form I created regarding this issue, it said .bin file type was not supported, which is what the save file type for 3d land is

**Comment by Jhon-ito** [comment]
What about a .zip file containing the .bin file. The author of this issue did that too.

**Comment by Mr-Mamma-Mia** [comment]
Unfortunately no matter where I look, I cannot find 2B8F, the closest I could could find 2B80, as all the other rows all had 0's at the end, so I have no clue what I'm doing wrong. I even sent my azahar save ti my 3ds to complete the rest of the levels from worlds 1 and 4 to get 2 stars on my save file, and still nothing

**Comment by Jhon-ito** [comment]
if you upload your save file as a compressed .zip file I can have a look at it.

**Comment by Mr-Mamma-Mia** [comment]
[Azahar saves (zip).zip](https://github.com/user-attachments/files/25343676/Azahar.saves.zip.zip)
here it is

**Comment by Jhon-ito** [comment]
[GameData.zip](https://github.com/user-attachments/files/25344830/GameData.zip)
Here, try this, it worked for me. I'm using Citra.

**Comment by Mr-Mamma-Mia** [comment]
OH MY GOD, TYSM, IT WORKED!!!!! Might I ask what value you changed and to what?

**Comment by Jhon-ito** [comment]
I just did the exact same steps I mentioned in my first post, I changed value 2B8F to 5C, maybe you were missing a step or maybe I wasn't clear enough with my instructions, I apologize if that's the case, english is not my first language.

**Comment by Mr-Mamma-Mia** [comment]
And question, was the value you changed 3C to 5C, and would you happen to know what it means when the square i guess turns yellow after modifying the value? Because that always happened to me when I changed to 5C

**Comment by Jhon-ito** [comment]
Yes, I changed it from 3C to 5C. Sorry I ddn't understand, what square are you talking about? in the hex editor? I'm using HxD

**Comment by Mr-Mamma-Mia** [comment]
The hex editor that I used whenever I changed a value, the area the value was in, aka the 2B80 row and the 0F column, changed to be yellow, so I'm not sure if that changed anything

**Comment by Jhon-ito** [comment]
Mine changes color to red when I modify any value, and, as I said in my post, then you have to save and exit the editor. The change of color may just mean unsaved changes in the file. At least that's what it means in the one I'm using.

**Comment by Mr-Mamma-Mia** [comment]
Hmm, weird, because I saved the file as a different name than the original to differentiate and then saved that in the SGE

**Comment by Louiskovski** [comment]
**I've found a much easier solution!**

When the title screen appears, press the A button repeatedly as quickly as possible. If it works, you'll access the save file menu!

It's best to use turbo button and a slow emulation setting. That should increase your chances of success.

BUT, to save from there on, **you must always save within the normal world, NEVER within the special world!** So, on the world map, always go back to the normal world through the pipe, press pause, and then tap save.

Once you've saved within the special world and restart the game, this trick won't work anymore.


I haven't tested it further, but it could be possible that the trick will also stop working if you get more stars on a save file. So, it's best to create a save state every now and then.

**Comment by tkdkssko-code** [comment]
How can I set up Android Azahar slowly?

**Comment by PlayerGameSK** [comment]
> How can I set up Android Azahar slowly?

Why are you asking that here lol @tkdkssko-code 

EDIT: Oh, I assume you mean how to slow it down, in which case I wouldn't know...
